app/Controllers/Approval.php

```diff
@@ -169,6 +171,43 @@ public function getPendingSpeakers(): ResponseInterface
         return $this->getPendingRoleEntries(Role::SPEAKER);
     }
 
+    public function canReject(int $userId, int $eventId): ResponseInterface
+    {
+        $canBeRejected = $this->checkIfSpeakerCanBeRejected($userId, $eventId);
+        if ($canBeRejected instanceof ResponseInterface) {
+            return $canBeRejected;
+        }
+        return $this->response->setJSON(['can_reject' => $canBeRejected]);
+    }
+
+    public function reject(int $userId, int $eventId): ResponseInterface
+    {
+        $canBeRejected = $this->checkIfSpeakerCanBeRejected($userId, $eventId);
+        if ($canBeRejected instanceof ResponseInterface) {
+            return $canBeRejected;
+        }
+        if (!$canBeRejected) {
+            return $this->response->setJSON(['error' => 'SPEAKER_CANNOT_BE_REJECTED'])->setStatusCode(400);
+        }
+
+        $speakerModel = model(SpeakerModel::class);
+        $speakerEntries = $speakerModel->getAllForUserAndEvent($userId, $eventId);
+        $imageFilenames = array_column($speakerEntries, 'photo');
+
+        $error = false;
+        foreach ($imageFilenames as $imageFilename) {
+            if (!$this->deleteImage($imageFilename)) {
+                $error = true;
+            }
+        }
+
+        $speakerModel->deleteAllForEvent($userId, $eventId);
+        if ($error) {
+            return $this->response->setJSON(['error' => 'IMAGE_DELETION_FAILED'])->setStatusCode(500);
+        }
+        return $this->response->setStatusCode(204);
```

Comment:
There was this one PR where you introduced HTTP response codes.
Since it took me some time to review this it might be good to check this PR in terms of HTTP response codes.

I will comment this only here but I think this will be a topic across this PR.

Reply:
I will have to rebase this PR now anyway. During that, I will hopefully also catch all remaining status codes.